### PR TITLE
added a checkUrl function to make sure that the DOM has been mounted

### PR DESCRIPTION
**Pull request by RichardHpa**
Currently the cypress suites fail constantly because of Cypress and React Hydrating the document. 

Issue: https://github.com/remix-run/indie-stack/issues/130

After looking through some of the cypress issues I came across this comment.
https://github.com/cypress-io/cypress/issues/7306#issuecomment-1152752612
It talks about adding a new cypress command which checks too makes sure that the url you are visiting matches the pathname.

Its not the best but since adding it in I have ran the cypress suite a dozen times with no fails

### CODE REVIEW

File: cypress/support/commands.ts [hunk 1]
```diff
@@ -71,6 +81,17 @@ function deleteUserByEmail(email: string) {
...
 Cypress.Commands.add("cleanupUser", cleanupUser);
 
+// We're waiting a second because of this issue happen randomly
+// https://github.com/cypress-io/cypress/issues/7306
+// Also added custom types to avoid getting detached
+// https://github.com/cypress-io/cypress/issues/7306#issuecomment-1152752612
+// ===========================================================
+function visitAndCheck(url: string) {
+  cy.visit(url);
+  cy.location("pathname").should("contain", url).wait(1000);
```
Comment (kentcdodds) [inline cypress/support/commands.ts:91]: Is 1000 milliseconds necessary? Seems long 🤔

**Review by kentcdodds — COMMENTED** [review]
Thank you! This will help a lot. One question.

File: cypress/support/commands.ts (hunk 1 above)
Comment (RichardHpa) [reply cypress/support/commands.ts:91]: Il have a play around and try find the shortest time we can use. Or could also just have a sending parameter for wait time.

Comment (kentcdodds) [reply cypress/support/commands.ts:91]: I wonder whether there's a way to turn this into something generic: https://github.com/cypress-io/cypress/issues/7306#issuecomment-1154997447 🤔

Comment (RichardHpa) [reply cypress/support/commands.ts]: I did try the `if (Cypress.dom.isDetached($el)) cy.wait(500);` But couldn't get it to work constantly which was annoying.

I shorted the wait to `100` but also made it someone could add in their own if they need longer.

**Comment by kentcdodds** [comment]
Rats. Looks like on CI that's not long enough :( Let's stick with 1000 as the default 😭 Cypress is supposed to make it so we don't have to do stuff like this. At least they're actively working on it.

**Comment by RichardHpa** [comment]
> Rats. Looks like on CI that's not long enough :( Let's stick with 1000 as the default 😭 Cypress is supposed to make it so we don't have to do stuff like this. At least they're actively working on it.

Done. Though I did keep it that you can change it if need be. Hopefully the Cypress team does do a fix for this soon.

If your happy with this il spin up other PR's in the other 2 offical stacks

**Review by kentcdodds — APPROVED** [review]
Thank you very much for this! Yes, I'd appreciate it if you apply this to the other two stacks. Thanks again 👏👏

**Comment by RichardHpa** [comment]
Created new PRs in each of the other 2 stacks

https://github.com/remix-run/grunge-stack/pull/92
https://github.com/remix-run/blues-stack/pull/104

**Pull request merged**